cmake/linker/armlink/target.cmake

```diff
@@ -55,7 +56,7 @@ macro(configure_linker_script linker_script_gen linker_pass_define)
       -P ${ZEPHYR_BASE}/cmake/linker/armlink/scatter_script.cmake
   )
 
-  if("LINKER_ZEPHYR_FINAL" IN_LIST "${linker_pass_define}")
+  if("LINKER_ZEPHYR_FINAL" IN_LIST linker_pass_define_list)
```

Comment:
Can you confirm that

IN_LIST linker_pass_define)

does not work?

I see that unexpectedly IN_LIST only works on variables. But I don't see what configure_file has to do with this limitation in IN_LIST.